### PR TITLE
Improve map with query expression code action

### LINKED ISSUE #38298: [Bug]: 'Map with a query expression' code action does not support for field access expressions
### Steps to Reproduce

Assign an array from a field access expression and find the quick actions

### Affected Version(s)

2201.2.2 RC1

### OS, DB, other environment details and versions

-

### Related area

-> Other Area

### Related issue(s) (optional)

_No response_

### Suggested label(s) (optional)

_No response_

### Suggested assignee(s) (optional)

_No response_

### PR DESCRIPTION
**Pull request by malinthar**
## Purpose
$subject

Fixes #38298 

## Samples
query_expression

## Check List 
- [ ] Read the [Contributing Guide](https://github.com/ballerina-platform/ballerina-lang/blob/master/CONTRIBUTING.md)
- [ ] Updated Change Log
- [ ] Checked Tooling Support (#<Issue Number>)
- [ ] Added necessary tests
  - [ ] Unit Tests
  - [ ] Spec Conformance Tests
  - [ ] Integration Tests
  - [ ] Ballerina By Example Tests
- [ ] Increased Test Coverage   
- [ ] Added necessary documentation  
  - [ ] API documentation 
  - [ ] Module documentation in Module.md files
  - [ ] Ballerina By Examples

### CODE REVIEW

File: language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/codeaction/providers/ConvertToQueryExpressionCodeAction.java [hunk 1]
```diff
@@ -176,65 +180,73 @@ private Optional<LhsRhsSymbolInfo> getLhsAndRhsSymbolInfo(NonTerminalNode matche
...
-            rhsNode = specificFieldNode.valueExpr().get();
-            lhsSymbol = semanticModel.symbol(lhsNode)
-                    .filter(symbol -> symbol.kind() == SymbolKind.RECORD_FIELD);
-            lhsType = lhsSymbol
-                    .map(symbol -> ((RecordFieldSymbol) symbol).typeDescriptor());
-        }
-        
+            matchedNode = matchedNode.parent();
+        } while (matchedNode.kind() != SyntaxKind.LOCAL_VAR_DECL ||
+                matchedNode.kind() != SyntaxKind.ASSIGNMENT_STATEMENT
```
Comment (IMS94) [inline language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/codeaction/providers/ConvertToQueryExpressionCodeAction.java:230]: Shouldn't we add a check like `matchedNode != null` too?

**Review by IMS94 — CHANGES_REQUESTED** [review]

File: language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/codeaction/providers/ConvertToQueryExpressionCodeAction.java (hunk 1 above)
Comment (malinthar) [reply language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/codeaction/providers/ConvertToQueryExpressionCodeAction.java:230]: If the matchedNode was null the code action would not be invoked. See https://github.com/ballerina-platform/ballerina-lang/blob/05c1101887657fe5cf4b5119676fb74e8c4cded1/language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/codeaction/CodeActionRouter.java#L81

Comment (IMS94) [reply language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/codeaction/providers/ConvertToQueryExpressionCodeAction.java:230]: But we modify that inside the `while` loop `matchedNode = matchedNode.parent();` right?

Comment (malinthar) [reply language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/codeaction/providers/ConvertToQueryExpressionCodeAction.java:230]: +1 added the the check

**Review by IMS94 — DISMISSED** [review]

**Review by IMS94 — DISMISSED** [review]

**Review by IMS94 — APPROVED** [review]

**Pull request merged**